sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/datasketchesAggregates.scala

```diff
@@ -315,7 +316,7 @@ case class HllUnionAgg(
             union.update(sketch)
             Some(union)
           } catch {
-            case _: java.lang.Error =>
+            case _: SketchesArgumentException | _: java.lang.Error =>
```

Comment:
The addition of capturing `SketchesArgumentException` is to fix the test failure for `hll.sql`.

```
diff --git a/sql/core/src/test/resources/sql-tests/results/hll.sql.out b/sql/core/src/test/resources/sql-tests/results/hll.sql.out
index adf6c40d1bd..8f4d14fcf8d 100644
--- a/sql/core/src/test/resources/sql-tests/results/hll.sql.out
+++ b/sql/core/src/test/resources/sql-tests/results/hll.sql.out
@@ -220,14 +220,8 @@ SELECT hll_sketch_estimate(CAST ('abc' AS BINARY))
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_sketch_estimate`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
 
 
 -- !query
@@ -235,14 +229,8 @@ SELECT hll_union(CAST ('abc' AS BINARY), CAST ('def' AS BINARY))
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_union`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
 
 
 -- !query
@@ -251,14 +239,8 @@ FROM (SELECT CAST('abc' AS BINARY) AS buffer)
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_union_agg`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
```